src/actions/SupplierCreditActions.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2020
+ */
+
+import { batch } from 'react-redux';
+import { createRecord, UIDatabase } from '../database';
+import { refreshData } from '../pages/dataTableUtilities/actions/tableActions';
+import { ROUTES } from '../navigation/constants';
+
+export const SUPPLIER_CREDIT_ACTIONS = {
+  CREATE_FROM_ITEM: 'SupplierCredit/createFromItem',
+  CLOSE: 'SupplierCredit/close',
+  SORT: 'SupplierCredit/sort',
+  EDIT_RETURN_AMOUNT: 'SupplierCredit/editReturnAmount',
+};
+
+const sort = sortKey => ({ type: SUPPLIER_CREDIT_ACTIONS.SORT, payload: { sortKey } });
+
+const close = () => ({ type: SUPPLIER_CREDIT_ACTIONS.CLOSE });
+
+const create = () => (dispatch, getState) => {
+  const { supplierCredit, user } = getState();
+  const { batches } = supplierCredit;
+  const { currentUser } = user;
+
+  // Only work with the batches whose return amount is greater than 0.
+  const batchesToReturn = batches.reduce(
```

Comment:
Possibly a better use case for `.filter` than `.reduce`?

Reply:
🤣 1bacf249a443c9826b189b8474a50df4b7a3c015